### PR TITLE
Add hostAliases in Master and Worker Pods

### PR DESCRIPTION
**Pull request by cdmikechen**
Add hostAliases to master statefulset.yaml and worker daemonset.yaml . Sometimes, Hadoop hosts is not configured in k8s cluster nodes or in DNS, so that we need to add `hostAliases` to let alluxio to identify Hadoop's hostname.

### CODE REVIEW

File: integration/kubernetes/helm-chart/alluxio/CHANGELOG.md [hunk 1]
```diff
@@ -159,3 +159,7 @@
 0.6.16
 
 - Change helm-chart fuse hostPath type from File to CharDevice
+
+0.6.17
+
+- Add hostAliases in Master and Worker Pods
```
Comment (jiacheliu3) [inline integration/kubernetes/helm-chart/alluxio/CHANGELOG.md:165]: @apc999 Just double checking, FUSE only talks to the Alluxio masters and workers right? When Alluxio(in K8s) is using HDFS(outside K8s on physical nodes, NAT provided from `hostAliases`) as UFS, the FUSE pods do NOT talk to HDFS in any way right?

Comment (jiacheliu3) [reply integration/kubernetes/helm-chart/alluxio/CHANGELOG.md:165]: The `hostAliases` are only added to the masters and workers. This means if the FUSE pods want to talk to the HDFS, they won't be able to resolve those IPs.

File: integration/kubernetes/helm-chart/alluxio/values.yaml [hunk 2]
```diff
@@ -436,3 +436,11 @@ logserver:
   medium: ""
   size: 4Gi
 
+# The pod's HostAliases. HostAliases is an optional list of hosts and IPs that will be injected into the pod's hosts file if specified.
+# It is mainly to provide the external host address for HDFS.
```
Comment (jiacheliu3) [inline integration/kubernetes/helm-chart/alluxio/values.yaml:440]:
```suggestion
# It is mainly to provide the external host addresses for services not in the K8s cluster, like HDFS.
```

File: integration/kubernetes/helm-chart/alluxio/templates/master/statefulset.yaml [hunk 3]
```diff
@@ -59,6 +59,16 @@ spec:
...
+      {{ if .Values.hostAliases }}
+      hostAliases:
+        {{- range .Values.hostAliases }}
+        - ip: {{ .ip }}
+          hostnames:
+            {{- range .hostnames }}
+            - {{ . }}
+            {{- end }}
+        {{- end }}
+      {{ end }}
```
Comment (jiacheliu3) [inline integration/kubernetes/helm-chart/alluxio/templates/master/statefulset.yaml:71]: This part is the same code as the one in worker YAML. You can extract this shared code into https://github.com/Alluxio/alluxio/blob/master/integration/kubernetes/helm-chart/alluxio/templates/_helpers.tpl

**Review by jiacheliu3 — COMMENTED** [review]
@cdmikechen Thanks for the improvement! The idea and design all look good to me. I left some comments regarding the code structure.

File: integration/kubernetes/helm-chart/alluxio/templates/master/statefulset.yaml (hunk 3 above)
Comment (cdmikechen) [reply integration/kubernetes/helm-chart/alluxio/templates/master/statefulset.yaml:71]: @jiacheliu3 OK~ I'll move these to `_helpers.tpl`

File: integration/kubernetes/helm-chart/alluxio/CHANGELOG.md (hunk 1 above)
Comment (cdmikechen) [reply integration/kubernetes/helm-chart/alluxio/CHANGELOG.md:165]: @jiacheliu3  I've tested in a fuse pod without `hostAliases`. I've created a new folder in `/mnt/alluxio-fuse/` and create a new file to this folder. It looks like except for a root permission error at the beginning, it is normal to create folders and files and write contents.

**Review by jiacheliu3 — APPROVED** [review]
@cdmikechen Thanks for the improvement! LGTM

**Comment by jiacheliu3** [comment]
Test failures are flakiness:
```
Error: 4.582 [ERROR] Tests run: 5, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.75 s <<< FAILURE! - in alluxio.master.transport.GrpcMessagingTransportTest
Error: 4.587 [ERROR] alluxio.master.transport.GrpcMessagingTransportTest.testEstablishConnection  Time elapsed: 0.098 s  <<< ERROR!
java.util.concurrent.ExecutionException: java.lang.NullPointerException: not on a Grpc messaging thread
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
```

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**